diff --git a/CTFd/auth.py b/CTFd/auth.py
index 0857bc20..a217ae0f 100644
--- a/CTFd/auth.py
+++ b/CTFd/auth.py
@@ -149,7 +149,7 @@ def reset_password(data=None):
         )
 
     if request.method == "POST":
-        email_address = request.form["email"].strip()
+        email_address = request.form["email"].strip().lower()
         user = Users.query.filter_by(email=email_address).first()
 
         get_errors()
